ir/src/ir2/graph/nodes.rs

```diff
@@ -0,0 +1,236 @@
+use crate::graph::{BackLink, Child, Graph, Leaf, Link, Node, Parent};
+use std::fmt::Debug;
+use std::ops::{Deref, DerefMut};
+
+#[derive(Clone, Eq, PartialEq)]
+pub enum NodeTypes {
+    Graph(Graph),
+    Node(Node),
+    I32(Leaf<i32>),
```

Comment:
Would this be Value(Leaf<MirValue>) basically?
And you'd have something like Variable(Leaf<MirVariable>) along side it as the two Leaf types?

Reply:
Yes pretty much, the Leaf is just used as a special case where it has no children. It's meant as a way to represent raw values, types, etc. Elements that cannot have sub-nodes essentially. I found it easier to think of it as a link to an arbitrary struct that should not link to other nodes (other than their parent).

Variable probably can be one too, but I need to think about edgecases more before committing to that answer. As a first guess, I agree.

The I32 naming is a relique from my testing, I'll rename to Value.